javascript/ql/lib/semmle/javascript/security/dataflow/DomBasedXssCustomizations.qll

```diff
@@ -335,4 +328,16 @@ module DomBasedXss {
 
   /** Gets the flow-label representing tainted values where the prefix is attacker controlled. */
   PrefixString prefixLabel() { any() }
+
+  /**
+   * A sanitizer that blocks the `PrefixString` label when the start of the string is being tested as being of a particular prefix.
+   */
+  abstract class PrefixStringSanitizer extends TaintTracking::SanitizerGuardNode,
+    TaintTracking::LabeledSanitizerGuardNode instanceof StringOps::StartsWith {
```

Comment:
Is there a need to extend both `SanitizerGuardNode` and `LabeledSanitizerGuardNode`?

Reply:
Probably not :+1:

Reply:
Yep, the tests still pass.